app/controllers/bookings_controller.rb

```diff
@@ -52,7 +52,7 @@ def decline
 
   def edit
     @booking = Booking.find(params[:id])
-    @activity = Activity.find(params[:activity_id])
+    @activity = @booking.activity
```

Comment:
@chloechemin This should fix the Edit bookings path, it was working sometimes because the number of bookings didn't exceed the activity.count, but as soon as it does it will raise an error saying eg. 'activity.id doesn't exist'.
Using the params I found that the activity_id was being created instead of referenced, so this line should resolve it!